python/pyiceberg/table/metadata.py

```diff
@@ -42,55 +45,79 @@
 from pyiceberg.typedef import EMPTY_DICT, IcebergBaseModel, Properties
 from pyiceberg.utils.datetime import datetime_to_millis
 
-CURRENT_SNAPSHOT_ID = "current_snapshot_id"
-CURRENT_SCHEMA_ID = "current_schema_id"
+CURRENT_SNAPSHOT_ID = "current-snapshot-id"
+CURRENT_SCHEMA_ID = "current-schema-id"
 SCHEMAS = "schemas"
-DEFAULT_SPEC_ID = "default_spec_id"
-PARTITION_SPEC = "partition_spec"
-PARTITION_SPECS = "partition_specs"
-SORT_ORDERS = "sort_orders"
+DEFAULT_SPEC_ID = "default-spec-id"
+PARTITION_SPEC = "partition-spec"
+PARTITION_SPECS = "partition-specs"
+SORT_ORDERS = "sort-orders"
+LAST_PARTITION_ID = "last-partition-id"
+LAST_ASSIGNED_FIELD_ID = "last-assigned-field-id"
 REFS = "refs"
+SPEC_ID = "spec-id"
+FIELD_ID = "field-id"
+FIELDS = "fields"
 
 INITIAL_SEQUENCE_NUMBER = 0
 INITIAL_SPEC_ID = 0
 DEFAULT_SCHEMA_ID = 0
 
 
-def check_schemas(values: Dict[str, Any]) -> Dict[str, Any]:
+def cleanup_snapshot_id(data: Dict[str, Any]) -> Dict[str, Any]:
+    """Runs before validation."""
+    if CURRENT_SNAPSHOT_ID in data and data[CURRENT_SNAPSHOT_ID] == -1:
+        # We treat -1 and None the same, by cleaning this up
+        # in a pre-validator, we can simplify the logic later on
+        data[CURRENT_SNAPSHOT_ID] = None
+    return data
+
+
+def check_schemas(table_metadata: TableMetadata) -> TableMetadata:
```

Comment:
Changed a lot of these validators to `after`, instead of `before`. The `before` would validate on the Python dictionaries without any alias'es applied. The `after` uses the metadata object itself, which is more safe and less error prone.

Reply:
This is also because the validators are refactored quite a bit.

Reply:
The `before` accept an dict with the unprocessed entries, the `after` an actual metadata object. I've switched the ordering because I think it makes sense to first have the `before`'s, and then the `after`'s

Reply:
I like how we use the same keys now.

Reply:
Yes, and it also handles the aliases. For example, for the `NestedField` we both accept `field_id` and `id` as the `id`. But since `id` is a reserved keyword we map `id` to `field_id`. In the `after` validator, we make sure that `id` is converted to `field_id`, which makes it less error-prone :)